packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -1504,3 +1500,25 @@ function isNullable(optionalValue: spec.OptionalValue | undefined): boolean {
         || (spec.isPrimitiveTypeReference(optionalValue.type)
             && optionalValue.type.primitive === spec.PrimitiveType.Any);
 }
+
+function paramJavadoc(name: string, optional?: boolean, summary?: string): string {
+    let documentation;
+    if (summary) {
+        documentation = summary;
+    } else {
+        documentation = '';
+    }
+
+    if (documentation.length > 0) {
+        if (!documentation.endsWith('.')) {
+            documentation += '.';
+        }
+        documentation += " ";
```

Comment:
This space gets added even if the "required" suffix isn't added later on.

Feels like this method could be simplified to something like:


```ts
function paramJavadoc(name: string, optional?: boolean, summary?: string): string {
  const parts = ['@param', name];
  if (summary) { parts.push(endWithPeriod(summary)); }
  if (!optional) { parts.push('This parameter is required.'); }

  return parts.join(' ');
}
```